src/QMCWaveFunctions/BsplineFactory/SplineR2R.cpp

```diff
@@ -110,10 +112,16 @@ void SplineR2R<ST>::applyRotation(const ValueMatrix& rot_mat, bool use_stored_co
         const auto index = i * Nsplines + k;
         newval += *(spl_coefs + index) * tmpU[k][j];
       }
-      *(spl_coefs + cur_elem) = newval;
+      new_coefs[cur_elem] = newval;
     }
   }
 
+  // Update the coefs
+  for (auto i = 0; i < coefs_tot_size; i++)
+  {
+    *(spl_coefs + i) = new_coefs[i];
```

Comment:
any reason not to use:
```c++
std::copy(new_coefs.begin(), new_coefs.end(), spl_coefs);
```
This would allow dropping the coefs_tot_size variable as well.

Reply:
Done.